src/client/pages/Embark/index.tsx

```diff
@@ -237,11 +241,24 @@ const useCreateQuoteCartId = ({ skip = false }) => {
     { data, error },
   ] = useCreateOnboardingQuoteCartMutation()
 
-  const createQuoteCart = useCallback(() => {
-    createOnboardingQuoteCart({
+  const [addCampaignCode] = useAddCampaignCodeMutation()
+
+  const createQuoteCart = useCallback(async () => {
+    const result = await createOnboardingQuoteCart({
       variables: { market: apiMarket, locale: isoLocale },
     })
-  }, [createOnboardingQuoteCart, apiMarket, isoLocale])
+    const quoteCartId = result.data?.onboardingQuoteCart_create.id
+    const savedCampaignCode = CampaignCode.get()
+
+    if (quoteCartId && savedCampaignCode !== null) {
```

Comment:
Remember to remove the `savedCampaignCode` in case `addCampaignCode` mutation fails

Reply:
Great!